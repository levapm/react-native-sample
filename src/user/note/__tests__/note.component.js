import React from 'react';
import { Alert } from 'react-native';
import { render, fireEvent } from 'react-native-testing-library';

jest.mock('Alert', () => {
    return {
        alert: jest.fn()
    }
});

import Note from '../note.component';

const NoteMock = {
    createdAt: new Date(),
    content: 'this is a test note',
    targetUserId: 'asdfasdfasdf',
    authorId: 'asdfasdfasdf',
    storeId: 'asdfasdfasdf',
};

describe('<Note />', () => {
    it('Renders content and button', () => {
        const tree = render((<Note note={NoteMock} onRemove={() => null} />)).toJSON();
        expect(tree.children.length).toMatchSnapshot();
    });

    it('Renders content and button', () => {
        const mockRemove = jest.fn();
        const { getByTestId } = render((<Note note={NoteMock} onRemove={mockRemove} />));

        fireEvent(getByTestId('noteRemoveButton'), 'onPress');
        expect(Alert.alert).toHaveBeenCalled();
        
        // manually click the second button
        Alert.alert.mock.calls[0][2][1].onPress();
        expect(mockRemove).toHaveBeenCalled();
    });
});
