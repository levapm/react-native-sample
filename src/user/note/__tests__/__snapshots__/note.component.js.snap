// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Note /> Renders content and button 1`] = `2`;
